File: testprojects/scala_main/pipelines/LIVY_SCALA/code/src/main/scala/io/prophecy/pipelines/livy_scala/graph/SQLStatement_1.scala
package io.prophecy.pipelines.livy_scala.graph

import io.prophecy.libs._
import io.prophecy.pipelines.livy_scala.config.ConfigStore._
import io.prophecy.pipelines.livy_scala.config.Context
import io.prophecy.pipelines.livy_scala.udfs.UDFs._
import io.prophecy.pipelines.livy_scala.udfs._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object SQLStatement_1 {

  def apply(
    context: Context,
    input_0: DataFrame
  ): (DataFrame, DataFrame, DataFrame, DataFrame) = {
    input_0.createOrReplaceTempView("input_0")
    (context.spark.sql("select * from input_0 where cast(year as int) >= 2011"),
     context.spark.sql(
       "select * from input_0 where industry_code_ANZSIC like '%A%'"
     ),
     context.spark.sql("select * from input_0 where variable like '%Total%'"),
     context.spark.sql("select * from input_0 where cast(value as int) > 600")
    )
  }

}

File: testprojects/python_main/pipelines/LIVY_PYTHON/code/livy_python/graph/SQLStatement_1.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.libs import typed_lit
from livy_python.config.ConfigStore import *
from livy_python.udfs.UDFs import *

def SQLStatement_1(
        spark: SparkSession,
        input_0: DataFrame, 
        in1: DataFrame, 
        in2: DataFrame
) -> (DataFrame, DataFrame, DataFrame):
    input_0.createOrReplaceTempView("input_0")
    in1.createOrReplaceTempView("in1")
    in2.createOrReplaceTempView("in2")
    df1 = spark.sql(
        "select input_0.year,input_0.variable,in1.value,input_0.unit,input_0.c_config from input_0,in1 where input_0.industry_code_ANZSIC=in1.industry_code_ANZSIC"
    )
    df2 = spark.sql("select * from in1 where industry_code_ANZSIC like '%A%'")
    df3 = spark.sql("select * from in2 where variable like '%Total%'")

    return df1, df2, df3
